### PR TITLE
Update Hibernate ORM, Hibernate Reactive

### LINKED ISSUE #16450: Accessing EntityManager prevents loading of lazy entities
## Describe the bug
Entity A has OneToOne to entity B, which has OneToMany to entity C. Iterating through all A's, B's and all C's.
(see attached project below)

```
    @Test
    @DataSet(value = "org/repro/ReproTest.yml")
    @Transactional // this annotation is required to reproduce the problem
    public void reproducer_SanityCheck() {
        // we have one A, with one B, with one C
        List<A> aList = A.listAll();
        assertThat(aList.size()).isEqualTo(1);

        A a = aList.get(0);
        B b = a.getB();
        assertThat(b).isNotNull();

        int numBs = b.getCList().size();
        assertThat(numBs).isEqualTo(1);
    }

    @Test
    @DataSet(value = "org/repro/ReproTest.yml")
    @Transactional // this annotation is required to reproduce the problem
    public void reproducer_Case1() {
        List<A> aList = A.listAll();
        assertThat(aList.size()).isEqualTo(1);

        A a = aList.get(0);
        B b = a.getB();
        assertThat(b).isNotNull();

        // after we do a listAll() on UX...
        List<UX> list = UX.listAll();

        // ... it seems as if B does not have any C
        int numCs = b.getCList().size();
        System.out.println("numBs: " + numCs);
        assertThat(numCs).isEqualTo(1);
    }
```

### Expected behavior
reproducer_Case1() should pass.

Entities should be retrieved/listed like this:
A1, B1, C1

### Actual behavior

... but this happens:
reproducer_Case1() fails... like C1 does not exist:
A1, B1, (access EM here), [C1 dissapears]

## To Reproduce
See attached project with test case. Run tests in ReproTest.

### Output of `uname -a` or `ver`
```
Linux a 5.11.10-1-MANJARO #1 SMP PREEMPT Thu Mar 25 17:14:55 UTC 2021 x86_64 GNU/Linux
```

### Output of `java -version`
```
openjdk version "11" 2018-09-25
OpenJDK Runtime Environment 18.9 (build 11+28)
OpenJDK 64-Bit Server VM 18.9 (build 11+28, mixed mode)
```

### Quarkus version or git rev
1.13.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)
```
Apache Maven 3.6.3 (NON-CANONICAL_2019-11-27T20:26:29Z_root)
Maven home: /opt/maven
Java version: 11, vendor: Oracle Corporation, runtime: /opt/openjdk-11
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.11.10-1-manjaro", arch: "amd64", family: "unix"
```

## Additional context
If the Transactional annotation is removed, the tests pass. But I need Transactional.
It seems that it does not matter if accessing EM is done through Panache or through EM directly (test reproducer_Case3_NoPanache()). 
In certain cases (in the original source code within the real project) listing specific entities causes the problem, while listing entities of other types does not. At first I thought that the entities need a relation to B, but it seems that is not the case. That's why there are two similar test: one with entity UX (UnrelatedX) and one with RX (RelatedX)... does not matter in this setup, though.
The problem happened first in a REST resource.. I managed to create a test case from it.

The only 'workaround' I found is to 'pre-load' all B's before doing the listAll().. but this will cause other problems.

**Use this project to reproduce:**
[repro-case-no-panache.zip](https://github.com/quarkusio/quarkus/files/6302613/repro-case-no-panache.zip)

### LINKED ISSUE #16463: "could not prepare statement" ("Connection is closed") when reusing a session for another transaction after a rollback
## Describe the bug

This sequence of actions causes ORM to reuse a connection that was closed implicitly by Agroal, ultimately throwing an exception:

* Open session manually
* Open JTA transaction manually
* Do stuff in ORM => ORM opens a transaction
* Roll back the transaction => ORM should have closed the connection, but it didn't, so Agroal closes the transaction without telling it.
* Start another transaction
* Do stuff in ORM => ORM thinks the connection is still open, so it reuses it, resulting in an exception.

This happens in particular when mass indexing in Hibernate Search, because Hibernate Search rolls back transactions instead of committing them, since it only reads data.

### Expected behavior

No exception. ORM should close connections on rollback, at least with the default connection handling mode.

### Actual behavior

Mass indexing fails after the first batch, with errors like this:

```
2021-04-13 08:53:01,407 ERROR [org.hib.sea.eng.rep.imp.LogFailureHandler] (Hibernate Search - Mass indexing - FruitEntity - Entity loading - 8) HSEARCH000058: Exception occurred javax.persistence.PersistenceException: org.hibernate.exception.GenericJDBCException: could not prepare statement
Failing operation:
Loading and extracting entity data for entity 'FruitEntity' during mass indexing
: javax.persistence.PersistenceException: org.hibernate.exception.GenericJDBCException: could not prepare statement
	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:154)
	at org.hibernate.query.internal.AbstractProducedQuery.list(AbstractProducedQuery.java:1602)
	at org.hibernate.query.Query.getResultList(Query.java:165)
	at org.hibernate.query.criteria.internal.compile.CriteriaQueryTypeQueryAdapter.getResultList(CriteriaQueryTypeQueryAdapter.java:76)
	at org.hibernate.search.mapper.orm.massindexing.impl.IdentifierConsumerDocumentProducer.loadAndIndexList(IdentifierConsumerDocumentProducer.java:154)
	at org.hibernate.search.mapper.orm.massindexing.impl.IdentifierConsumerDocumentProducer.loadAndIndexAllFromQueue(IdentifierConsumerDocumentProducer.java:123)
	at org.hibernate.search.mapper.orm.massindexing.impl.IdentifierConsumerDocumentProducer.run(IdentifierConsumerDocumentProducer.java:103)
	at org.hibernate.search.util.common.impl.CancellableExecutionCompletableFuture$CompletingRunnable.run(CancellableExecutionCompletableFuture.java:49)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: org.hibernate.exception.GenericJDBCException: could not prepare statement
	at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:47)
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:113)
	at org.hibernate.engine.jdbc.internal.StatementPreparerImpl$StatementPreparationTemplate.prepareStatement(StatementPreparerImpl.java:186)
	at org.hibernate.engine.jdbc.internal.StatementPreparerImpl.prepareQueryStatement(StatementPreparerImpl.java:151)
	at org.hibernate.loader.Loader.prepareQueryStatement(Loader.java:2103)
	at org.hibernate.loader.Loader.executeQueryStatement(Loader.java:2040)
	at org.hibernate.loader.Loader.executeQueryStatement(Loader.java:2018)
	at org.hibernate.loader.Loader.doQuery(Loader.java:948)
	at org.hibernate.loader.Loader.doQueryAndInitializeNonLazyCollections(Loader.java:349)
	at org.hibernate.loader.Loader.doList(Loader.java:2849)
	at org.hibernate.loader.Loader.doList(Loader.java:2831)
	at org.hibernate.loader.Loader.listIgnoreQueryCache(Loader.java:2663)
	at org.hibernate.loader.Loader.list(Loader.java:2658)
	at org.hibernate.loader.hql.QueryLoader.list(QueryLoader.java:506)
	at org.hibernate.hql.internal.ast.QueryTranslatorImpl.list(QueryTranslatorImpl.java:400)
	at org.hibernate.engine.query.spi.HQLQueryPlan.performList(HQLQueryPlan.java:219)
	at org.hibernate.internal.SessionImpl.list(SessionImpl.java:1414)
	at org.hibernate.query.internal.AbstractProducedQuery.doList(AbstractProducedQuery.java:1625)
	at org.hibernate.query.internal.AbstractProducedQuery.list(AbstractProducedQuery.java:1593)
	... 12 more
Caused by: java.sql.SQLException: Connection is closed
	at io.agroal.pool.wrapper.ConnectionWrapper.lambda$static$0(ConnectionWrapper.java:51)
	at com.sun.proxy.$Proxy92.prepareStatement(Unknown Source)
	at io.agroal.pool.wrapper.ConnectionWrapper.prepareStatement(ConnectionWrapper.java:616)
	at org.hibernate.engine.jdbc.internal.StatementPreparerImpl$5.doPrepare(StatementPreparerImpl.java:149)
	at org.hibernate.engine.jdbc.internal.StatementPreparerImpl$StatementPreparationTemplate.prepareStatement(StatementPreparerImpl.java:176)
	... 28 more
```

## To Reproduce

Originally [reported by @suchwerk](https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/Hibernate.20Search.20-.20Massindexer): thanks to him!

Reproducer (in Quarkus): https://github.com/suchwerk/quarkus-hibernate-search-massindexer

You'll need to start an Elasticsearch container, then run the tests:

```
docker run -it --rm=true --name es-7.10-it -p 9200:9200 -e "discovery.type=single-node" docker.elastic.co/elasticsearch/elasticsearch-oss:7.10.0
./mvnw clean verify
```

### Configuration

N/A

### Screenshots

N/A

### Quarkus version or git rev

1.13.1.Final

Probably applies to 1.13.x only, since we switched the connection handling mode in 1.13.0.Final.

### LINKED ISSUE #16712: hibernate-reactive NullpointException in case Quarkus log in set to debug
## Describe the bug
hibernate-reactive NullpointException
When I set the log level to debug and run it, a nullpointexception occurs and the application does not work properly.

### Configuration
```properties
quarkus:
  log:
    level: DEBUG
  datasource:
    db-kind: postgresql
    username: postgres
    password: 1234
    reactive:
      url: postgresql://localhost:5432/postgres

```

### Screenshots
16:35:55 ERROR [io.qu.ru.Application] (Quarkus Main Thread) Failed to start application (with profile dev): java.lang.NullPointerException
	at org.hibernate.reactive.persister.collection.impl.ReactiveOneToManyPersister.getSQLInsertRowString(ReactiveOneToManyPersister.java:144)
	at org.hibernate.persister.collection.AbstractCollectionPersister.logStaticSQL(AbstractCollectionPersister.java:693)
	at org.hibernate.persister.collection.AbstractCollectionPersister.<init>(AbstractCollectionPersister.java:529)
	at org.hibernate.persister.collection.OneToManyPersister.<init>(OneToManyPersister.java:69)
	at org.hibernate.reactive.persister.collection.impl.ReactiveOneToManyPersister.<init>(ReactiveOneToManyPersister.java:52)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at org.hibernate.persister.internal.PersisterFactoryImpl.createCollectionPersister(PersisterFactoryImpl.java:152)
	at org.hibernate.persister.internal.PersisterFactoryImpl.createCollectionPersister(PersisterFactoryImpl.java:140)
	at org.hibernate.metamodel.internal.MetamodelImpl.initialize(MetamodelImpl.java:225)
	at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:303)
	at org.hibernate.reactive.session.impl.ReactiveSessionFactoryImpl.<init>(ReactiveSessionFactoryImpl.java:33)
	at io.quarkus.hibernate.reactive.runtime.boot.FastBootReactiveEntityManagerFactoryBuilder.build(FastBootReactiveEntityManagerFactoryBuilder.java:34)
	at io.quarkus.hibernate.reactive.runtime.FastBootHibernateReactivePersistenceProvider.createEntityManagerFactory(FastBootHibernateReactivePersistenceProvider.java:83)
	at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:80)
	at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:55)
	at io.quarkus.hibernate.orm.runtime.JPAConfig$LazyPersistenceUnit.get(JPAConfig.java:118)
	at io.quarkus.hibernate.orm.runtime.JPAConfig.startAll(JPAConfig.java:42)
	at io.quarkus.hibernate.orm.runtime.JPAConfig_Subclass.startAll$$superaccessor4(JPAConfig_Subclass.zig:650)
	at io.quarkus.hibernate.orm.runtime.JPAConfig_Subclass$$function$$4.apply(JPAConfig_Subclass$$function$$4.zig:29)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)
	at io.quarkus.arc.runtime.devconsole.InvocationInterceptor.proceed(InvocationInterceptor.java:63)
	at io.quarkus.arc.runtime.devconsole.InvocationInterceptor.monitor(InvocationInterceptor.java:51)
	at io.quarkus.arc.runtime.devconsole.InvocationInterceptor_Bean.intercept(InvocationInterceptor_Bean.zig:521)
	at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
	at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:32)
	at io.quarkus.hibernate.orm.runtime.JPAConfig_Subclass.startAll(JPAConfig_Subclass.zig:608)
	at io.quarkus.hibernate.orm.runtime.HibernateOrmRecorder.startAllPersistenceUnits(HibernateOrmRecorder.java:88)
	at io.quarkus.deployment.steps.HibernateOrmProcessor$startPersistenceUnits951856026.deploy_0(HibernateOrmProcessor$startPersistenceUnits951856026.zig:74)
	at io.quarkus.deployment.steps.HibernateOrmProcessor$startPersistenceUnits951856026.deploy(HibernateOrmProcessor$startPersistenceUnits951856026.zig:40)
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:762)
	at io.quarkus.runtime.Application.start(Application.java:90)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:100)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:66)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:42)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:119)
	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.runner.bootstrap.StartupActionImpl$3.run(StartupActionImpl.java:134)
	at java.base/java.lang.Thread.run(Thread.java:836)

### Output of `java -version`
jdk 11

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)
Apache Maven 3.6.3

### LINKED ISSUE #16860: Native runtime error due to HibernateBasicProxy
## Describe the bug
Hello,
When using a small hierarchy of JPA entities (abstract parent and concrete child plus discriminator), Hibernate generates a proxy entity that does not seem to be kept by the quarkus-hibernate-orm extension during native compilation.

### Expected behavior
The entity should be loaded from database when in native mode. This is the observed behavior with the uber-jar packaging.

### Actual behavior
When trying to retrieve an entity, the following error occurs:
```
Caused by: java.lang.InstantiationException: Type `org.example.quarkus.AbstractEntity$HibernateBasicProxy$SwF3dMZa` can not be instantiated reflectively as it does not have a no-parameter constructor or the no-parameter constructor has not been added explicitly to the native image.
at java.lang.Class.newInstance(DynamicHub.java:912)
at org.hibernate.bytecode.internal.bytebuddy.BasicProxyFactoryImpl.getProxy(BasicProxyFactoryImpl.java:57)
```

## To Reproduce
Please find attached a reproducer:
[quarkus-hibernateproxy-reproducer.zip](https://github.com/quarkusio/quarkus/files/6389443/quarkus-hibernateproxy-reproducer.zip)

1. Build the native executable
```
mvn clean package -Pnative
```
2. Start the external standalone H2 database
```
java -cp ./h2/h2-1.4.199.jar org.h2.tools.Server -ifNotExists &
```
3. Start the quarkus native executable 
```
./target/quarkus-hibernateproxy-reproducer-0.0.1-SNAPSHOT-runner
```
4. Access the GET /entities endpoint
```
curl localhost:8080/entities
```

### Configuration
This is the application.properties of the above reproducer. Real world usage is Oracle database (btw, many thanks for this new feature).
```properties
quarkus.datasource.db-kind=h2
quarkus.datasource.jdbc.url=jdbc:h2:tcp://localhost/./h2/quarkus
quarkus.datasource.jdbc.driver=org.h2.Driver
quarkus.hibernate-orm.dialect=org.hibernate.dialect.H2Dialect
quarkus.hibernate-orm.statistics=true
quarkus.hibernate-orm.log.sql=true
quarkus.hibernate-orm.database.generation=drop-and-create
quarkus.hibernate-orm.sql-load-script=import.sql
```
This configuration and the -ifNotExists flag on the h2 command line will result in the creation of a local database file
`quarkus-hibernateproxy-reproducer\h2\quarkus.mv.db` which can be deleted between test runs.

## Environment (please complete the following information):
Real world is OpenShift. Reproducer details follows.

### Output of `uname -a` or `ver`
`Linux rhel.domain.local 4.18.0-240.22.1.el8_3.x86_64 #1 SMP Thu Mar 25 14:36:04 EDT 2021 x86_64 x86_64 x86_64 GNU/Linux`

### Output of `java -version`
```
openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment GraalVM CE 21.1.0 (build 11.0.11+8-jvmci-21.1-b05)
OpenJDK 64-Bit Server VM GraalVM CE 21.1.0 (build 11.0.11+8-jvmci-21.1-b05, mixed mode, sharing)
```

### Quarkus version or git rev
1.13.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)
```
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /root/apache-maven-3.6.3
Java version: 11.0.11, vendor: GraalVM Community, runtime: /root/graalvm-ce-java11-21.1.0
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.18.0-240.22.1.el8_3.x86_64", arch: "amd64", family: "unix"
```

### native-image-agent
Running the uber-jar with the native-image-agent
```
java -agentlib:native-image-agent=config-output-dir=/quarkus/agent -jar target/quarkus-hibernateproxy-reproducer-0.0.1-SNAPSHOT-native-image-source-jar/quarkus-hibernateproxy-reproducer-0.0.1-SNAPSHOT-runner.jar
```
Generates a reflect-config.json file that contains reference to the culprit proxy class
```
$ grep -B1 -A2 BasicProxy /quarkus/agent/reflect-config.json
{
  "name":"org.example.quarkus.AbstractEntity$HibernateBasicProxy$SwF3dMZa",
  "methods":[{"name":"<init>","parameterTypes":[] }]
},
```
### zulip
https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/Native.20runtime.20error.20due.20to.20HibernateBasicProxy

### PR DESCRIPTION
**Pull request by Sanne**
Fixes #16860
Fixes #16450 
Fixes #16463
Fixes #16712

### CODE REVIEW

**Review by yrodiere — APPROVED** [review]

**Comment by yrodiere** [comment]
@Sanne @DavideD @gavinking Plenty of failing tests for Hibernate Reactive after the upgrade... I see timeouts in particular. Could this be related to the `session.close()` changes?

**Comment by Sanne** [comment]
I'm sorry, I don't know. I guess it could relate, but things were not being closed correctly even before the patch so while it's certainly a suspect, it doesn't feel too likely either

**Comment by DavideD** [comment]
Weird, wasn't it tested before?

@yrodiere Possible, but it would surprise me a bit. Closing we didn't have a reactive close in the previous version of Hibernate Reactive (even if the connection underneath was reactive) so connection were already closed closed the wrong way even before the update.

**Comment by DavideD** [comment]
Could it be related to the upgrade to ORM 5.5?
Other changes is the upgrade to Mutiny 0.16 and a bug fix to one-to-one associations

**Comment by yrodiere** [comment]
> What's the error anyway?

I don't know more than what appears on the report; apparently the integration test times out. So I think something hangs forever in Reactive. May be a problem in tests, though they used to pass and they didn't change...

See https://github.com/quarkusio/quarkus/runs/2475746313

**Comment by Sanne** [comment]
Ok I think I have a more concrete idea.  With the Mutiny API wrappers, an async method without any subscription is never invoked - this is different than a Future, in which case the lack of a subscription will defer the operation but it will still be run eventually.

So the previous version of Hibernate Reactive's `close()` implementation was:

```
    public void close() {
		delegate.close();
    }
```
Which was wrong as Vert.x 4 made the "close" asynchronous, but at least it would be triggered eventually.

But now that in the Mutiny version it was patched as:

```
	public Uni<Void> close() {
		return uni( delegate::reactiveClose );
	}
```

The problem is amplified in the Quarkus code as the CDI dispose method is implemented as:

```
public void disposeMutinySession(@Disposes Mutiny.Session reactiveSession) {
        if (reactiveSession != null) {
            reactiveSession.close();
        }
    }
```
N.B. the lack of subcriptions (not sure how to fix this in a CDI friendly way && without blocking)

So now the end effect is that since it's a Mutiny based wrapper and there is actually no subscription, it's never triggered and we run out of connections.

**Comment by Sanne** [comment]
Pushed something for CI to test - I'm having trouble testing with DB2 locally

**Comment by Sanne** [comment]
Soo still some failing tests but they seem unrelated to my changes. 

WDYT @yrodiere @gsmet ?
We need this, lots of progress is being stalled, so while I don't like CI status I'm inclined to merge

File: extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/ReactiveSessionProducer.java [hunk 1]
```diff
@@ -34,13 +34,24 @@
 
     public void disposeStageSession(@Disposes Stage.Session reactiveSession) {
         if (reactiveSession != null) {
-            reactiveSession.close();
+            //We're ignoring the returned CompletionStage!
```
Comment (yrodiere) [inline extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/ReactiveSessionProducer.java:37]: I agree with the change but I think the comment should be:

```suggestion
            //We must not ignore the returned CompletionStage!
```

File: extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/ReactiveSessionProducer.java [hunk 2]
```diff
@@ -34,13 +34,24 @@
...
+            //as it won't be able to return connections to other consumers when it's saturated,
+            //ensuring close operations are handled.
+            reactiveSession.close().toCompletableFuture().join();
         }
     }
 
     public void disposeMutinySession(@Disposes Mutiny.Session reactiveSession) {
         if (reactiveSession != null) {
-            reactiveSession.close();
+            //We're ignoring the returned CompletionStage!
```
Comment (yrodiere) [inline extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/ReactiveSessionProducer.java:48]: I agree with the change but I think the comment should be:

```suggestion
            //We must not ignore the returned CompletionStage!
```

**Review by yrodiere — APPROVED** [review]
LGTM apart from a comment that seems misleading. In any case, +1 to merge as the failures seem completely unrelated.

File: extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/ReactiveSessionProducer.java (hunk 1 above)
Comment (Sanne) [reply extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/ReactiveSessionProducer.java:37]: I'm actually referring to the fact that the `disposeStageSession` method itself (the one we're changing here) is returning void - while it *should* chain the event back to a reactive chain. That's sadly not possible with CDI at the moment.

So "must not ignore" would IMO be confusing as we are actually ignoring it here?

**Comment by Sanne** [comment]
> LGTM apart from a comment that seems misleading. In any case, +1 to merge as the failures seem completely unrelated.

right - and I've checked now that the other failures are indeed reproducible w/o my patch. Stuart also seems to have just fixed one of them in #16986 

I'll merge - thanks!

**Comment by Sanne** [comment]
Sadly, I've decided to remove the "backport" label:

Let's try to backport these fixes only if there's strong need - the problem is that while I'm comfortable with the changes in ORM, there is a high coupling requiring the HR upgrade as well - which now requires Vertx4 in this release.

Backporting would need an alternative stream of HR artifacts. cc/ @gavinking @DavideD

**Comment by gavinking** [comment]
Off the top of my head I don't think it would be that *hard* to do a dedicated HR release against the pre-vert.x 4 version. But i'm having a hard time gauging the *priority*. How important is this?

**Comment by Sanne** [comment]
@gavinking I think we can live just fine without one for now, I only mentioned it so you're aware that we might need one in the future.

We should also learn about such constraints, clearly I got the release sequence puzzle wrong this time - sorry

**Comment by gavinking** [comment]
OK. Let me know if you change your mind. I really don't think it would be a big deal to do a release.

**Comment by DavideD** [comment]
I didn't check the whole consversation, but releasing different versions of
hibernate reactive is pretty straight forward. For different versions of
vertx (3
9 and 4.0) we need to maintain different branches though.

**Pull request merged**